src/RuleSet/DeclarationBlock.php

```diff
@@ -21,14 +27,33 @@
  * matching elements.
  *
  * Declaration blocks usually appear directly inside a `Document` or another `CSSList` (mostly a `MediaQuery`).
+ *
+ * Note that `CSSListItem` extends both `Commentable` and `Renderable`, so those interfaces must also be implemented.
  */
-class DeclarationBlock extends RuleSet
+class DeclarationBlock implements CSSElement, CSSListItem, Positionable, RuleContainer
 {
+    use CommentContainer;
+    use Position;
+
     /**
      * @var array<Selector|string>
      */
     private $selectors = [];
 
+    /**
+     * @var RuleSet
+     */
+    private $ruleSet;
+
+    /**
+     * @param int<1, max>|null $lineNumber
+     */
+    public function __construct(?int $lineNumber = null)
+    {
+        $this->setPosition($lineNumber);
+        $this->ruleSet = new RuleSet($lineNumber);
```

Comment:
Nitpick: I usually prefer initializing the dependencies before setting properties in the constructor. That's a personal preference, though, so feel free to ignore it if you prefer the other order.

Reply:
The dependency (`Position`) _is_ initialized before the property (`$ruleSet`) here.

Hadn't really thought about it, but from C++ we always initialize the superclass first, which is partially enforced as a language construct.

Reply:
Independent of personal preferences, I'd like use to use some terms in compatible ways in this project. :-)

I know the term "dependency" (in the context classes in software architecture) to mean "one class uses/extends/implements another class"). That's also what dependency injection takes care of.

What's your understanding of the term, @JakeQZ ?

Reply:
> So am happy to switch the order :)

Now done, and rebased.